### ISSUE TITLE
Opt: Always start every spirv-opt run with CFGCleanup; always leave every pass with no unreachable blocks

### ISSUE THREAD

**Issue by greg-lunarg** [KhronosGroup/SPIRV-Tools#947]
First, this is more efficient as we will not be optimizing code that is not executable.

Second, this will make for simpler optimization algorithms as they will not have to take unreachable blocks into account.

It is recommended that, whenever possible, CFGCleanup be used to remove unreachable blocks at the end of passes.

**Comment by dneto0** [comment]
I don't think this should be on automatically.   The spirv-opt tool is a get-what-you-ask-for tool.  We're following the pattern of LLVM's `opt` tool.

I think it's perfectly fine for a transform pass to say "assumes cfg-cleanup has removed dead blocks".  Then it's in the control of the user to know when exactly that's required, and it puts the responsibility on them.

Minor point:  structured control flow allows merge-block and continue-target to be unreachable, even as it requires those blocks to be named (and exist).

**Comment by greg-lunarg** [comment]
So would it be ok for a pass to abort if it is given unreachable code?

**Comment by greg-lunarg** [comment]
BTW, assuming "unreachable"  continues and merges are ok as long as a merge statement references them.
